jekyll/_cci2/optimization-cookbook.md

```diff
@@ -0,0 +1,153 @@
+
+`orbs:
+  queue: eddiewebb/queue@1.1.2`
+
+4) Use `queue` elements in your existing workflows and jobs.
+
+5) Opt-in to use of third-party orbs on your organization’s **Security Settings** page.
+
+### Blocking Workflows
+
+One of the easiest ways to prevent workflow concurrency using the CircleCI Single-Threading orb is to enable "blocking" of any workflows with an earlier timestamp. By setting the `block-workflow` parameter value to `true`, all workflows will be forced to run consecutively, not concurrently, thereby limiting the number of workflows in the queue. In turn, this will also improve overall performance while making sure no workflows are discarded.
+
+{% raw %}
+
+```yaml
+Version: 2.1
+docker:
+  - image: 'circleci/node:10'
+parameters:
+  block-workflow:
+    default: true
+    description: >-
+      If true, this job will block until no other workflows with an earlier
+      timestamp are running. Typically used as first job.
+    type: boolean
+  consider-branch:
+    default: true
+    description: Should we only consider jobs running on the same branch?
+    type: boolean
+  consider-job:
+    default: false
+    description: Deprecated. Please see block-workflow.
+    type: boolean
+  dont-quit:
+    default: false
+    description: >-
+      Quitting is for losers. Force job through once time expires instead of
+      failing.
+    type: boolean
+  only-on-branch:
+    default: '*'
+    description: Only queue on specified branch
+    type: string
+  time:
+    default: '10'
+    description: How long to wait before giving up.
+    type: string
+  vcs-type:
+    default: github
+    description: Override VCS to 'bitbucket' if needed.
+    type: string
+steps:
+  - until_front_of_line:
+      consider-branch: <<parameters.consider-branch>>
+      consider-job: <<parameters.consider-job>>
+      dont-quit: <<parameters.dont-quit>>
+      only-on-branch: <<parameters.only-on-branch>>
+      time: <<parameters.time>>
+      vcs-type: <<parameters.vcs-type>>
+```
+{% endraw %}
+
+## Caching
+
+One of the quickest and easiest ways to optimize your builds and workflows is to implement specific caching strategies so you can use existing data from previous builds and workflows. Whether you choose to use a package management application (e.g. Yarn, Bundler, etc), or manually configure your caching, utilizing the best and most effective caching strategy may improve overall performance. In this section, several different use cases are described that may assist you in determining which caching method is best for your implementation.
+
+If a job fetches data at any point, it is likely that you can make use of caching. A common example is the use of a package/dependency manager. If your project uses Yarn, Bundler, or Pip, for example, the dependencies downloaded during a job can be cached for later use rather than being re-downloaded on every build. The example below shows how you can use a package manager for caching.
```

Comment:
> The example below shows how you can use a package manager for caching.

This sentence should maybe be reversed?

"The example below shows how you can use caching for a package manager."

Reply:
Fixed...thanks for the help on this one.